pentaho5/src/main/java/pt/webdetails/cpk/InterPluginBroker.java

```diff
@@ -45,49 +47,63 @@ public static void run( Map<String, Object> params, OutputStream out ) throws Ex
+      final String pluginId = CorePlugin.CDE.getId();
+      final String beanId = getSettingValue( CDE_RENDER_API_BEAN_ID_TAG, CDE_RENDER_API_BEAN_ID );
+      final String method = getSettingValue( CDE_RENDER_API_RENDER_METHOD_TAG, CDE_RENDER_API_RENDER_METHOD );
 
-    String response = pluginCall != null
-      ? pluginCall.call( parameters.getParameters() )
-      : "No valid InterPluginCall available";
+      logger.error( "No valid InterPluginCall available" +
+        "for the plugin: '" + pluginId + "', beanId: '" + beanId + "' and method: '" + method + "'." );
 
+      return;
+    }
+
+    String response = pluginCall.call( parameters.getParameters() );
     PluginIOUtils.writeOutAndFlush( out, response );
   }
 
   private static IPluginCall getPluginCall() {
-    if ( cdeRenderApiCall != null ) {
-      return cdeRenderApiCall;
+    if ( pluginCall != null ) {
+      return pluginCall;
     }
 
-    IPluginCall cdePluginCall;
-    final PluginEnvironment environment = PluginEnvironment.env();
     final String pluginID = CorePlugin.CDE.getId();
+    final PluginEnvironment environment = PluginEnvironment.env();
 
     // 1. try configured values
-    String beanId = PentahoSystem.getSystemSetting( CDE_RENDER_API_BEAN_ID_TAG, CDE_RENDER_API_BEAN_ID );
-    String method = PentahoSystem.getSystemSetting( CDE_RENDER_API_RENDER_METHOD_TAG, CDE_RENDER_API_RENDER_METHOD );
+    String beanId = getSettingValue( CDE_RENDER_API_BEAN_ID_TAG, "" );
+    String method = getSettingValue( CDE_RENDER_API_RENDER_METHOD_TAG, "" );
 
-    cdePluginCall = environment.getPluginCall( pluginID, beanId, method );
-    if ( beanExists( cdePluginCall ) ) {
-      return ( cdeRenderApiCall = cdePluginCall );
+    IPluginCall cdePluginCall = environment.getPluginCall( pluginID, beanId, method );
+    if ( cdePluginCall.exists() ) {
+      return ( pluginCall = cdePluginCall );
     }
 
     // 2. fallback to latest cde render bean id
     cdePluginCall = environment.getPluginCall( pluginID, CDE_RENDER_API_BEAN_ID, CDE_RENDER_API_RENDER_METHOD );
-    if ( beanExists( cdePluginCall ) ) {
-      return ( cdeRenderApiCall = cdePluginCall );
+    if ( cdePluginCall.exists() ) {
+      return ( pluginCall = cdePluginCall );
     }
 
     // 3. fallback to legacy cde render bean id
     cdePluginCall = environment.getPluginCall( pluginID, CDE_RENDER_API_LEGACY_BEAN_ID, CDE_RENDER_API_RENDER_METHOD );
-    if ( beanExists( cdePluginCall ) ) {
-      return ( cdeRenderApiCall = cdePluginCall );
+    if ( cdePluginCall.exists() ) {
+      return ( pluginCall = cdePluginCall );
     }
 
     return null;
   }
 
-  private static boolean beanExists( IPluginCall pluginCall ) {
-    return ( (InterPluginCall) pluginCall).beanExists();
+  private static String getSettingValue( String tag, String defaultValue ) {
+    final PluginSettings settings = PluginEnvironment.env().getPluginSettings();
+
+    String value = settings.getTagValue( tag ).get( 0 );
```

Comment:
I think we get an IndexOutOfBoundException here. You probably want to check  something lke this:
```
List<String> values = settings.getTagValue(tag);
return values.isEmpty() ? defaultValue : values.get(0);
```